### ISSUE TITLE
An issue for the JTT code.

### ISSUE THREAD

**Issue by LJSthu** [facebookresearch/BalancingGroups#5]
Sorry to disturb you. 
I wonder why the code for upweighting is like "self.weights[i] += predictions.detach() * (self.hparams["up"] - 1)". I think this will upweigh the correctly-classified samples.

**Comment by BadrYoubiIdrissi** [comment]
Hello,

You are absolutely right. This is a mistake. It was unfortunately introduced in the code cleanup just before release ! However, the final experiments of the paper were launched with the following line of code 

`self.weights[i] += predictions.argmax(1).ne(y).long().detach()*(self.hparams["up"] - 1)`

In this line, the variable `predictions` are the model logits. The difference with the current code is mainly .ne(y) vs .eq(y), with that line, the wrong predictions are upsampled, as intended by JTT.

The issue is now fixed ! Thank you for bringing it to our attention. I'd like to insist on the validity of the paper's results.

Here is the updated code 

https://github.com/facebookresearch/BalancingGroups/blob/1ac1b9db42c1c1a8c4cd2faa57194490140f6f05/models.py#L267-L272

**Comment by LJSthu** [comment]
Thanks!
